hopsworks-common/src/main/java/io/hops/hopsworks/common/serving/inference/LocalhostSkLearnInferenceController.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * This file is part of Hopsworks
+ * Copyright (C) 2018, Logical Clocks AB. All rights reserved
+ *
+ * Hopsworks is free software: you can redistribute it and/or modify it under the terms of
+ * the GNU Affero General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * Hopsworks is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ * PURPOSE.  See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.hops.hopsworks.common.serving.inference;
+
+import io.hops.common.Pair;
+import io.hops.hopsworks.common.dao.serving.Serving;
+import io.hops.hopsworks.common.util.Settings;
+import io.hops.hopsworks.restutils.RESTCodes;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.protocol.HttpContext;
+import org.apache.http.util.EntityUtils;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.ejb.ConcurrencyManagement;
+import javax.ejb.ConcurrencyManagementType;
+import javax.ejb.EJB;
+import javax.ejb.Singleton;
+import javax.ejb.TransactionAttribute;
+import javax.ejb.TransactionAttributeType;
+import javax.enterprise.inject.Alternative;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * SkLearn Localhost Inference Controller
+ *
+ * Sends inference requests to a local sklearn flask server to get a prediction response
+ */
+@Alternative
+@Singleton
+@TransactionAttribute(TransactionAttributeType.NEVER)
+@ConcurrencyManagement(ConcurrencyManagementType.BEAN)
+public class LocalhostSkLearnInferenceController implements SkLearnInferenceController {
+
+  @EJB
+  private Settings settings;
+
+  private static final Logger logger = Logger.getLogger(LocalhostSkLearnInferenceController.class.getName());
+
+  CloseableHttpClient httpClient = null;
+  PoolingHttpClientConnectionManager cm = null;
+  
+  /**
+   * Initialize the HTTP connection pool configuration
+   */
+  @PostConstruct
+  public void init() {
+    cm = new PoolingHttpClientConnectionManager(30l, TimeUnit.SECONDS);
```

Comment:
I don't think that SkLearn and TfServing should have 2 different connection pools. The `HttpClient` is thread safe. Ideally we should have a single HttpClient for the whole application, but that's another topic. 
SkLearn and TfServing should share the HttpClient/Connection Pool. This also has the benefit that we don't need to duplicate the parameters (pool size, max connections and so on...)

Reply:
Done